cmd/geth/main.go

```diff
@@ -400,6 +401,9 @@ func run(ctx *cli.Context) {
 
 	ethereum, err := eth.New(cfg)
 	if err != nil {
+		if datadirInUseErrRE.MatchString(err.Error()) {
+			err = fmt.Errorf("%v (another instance of geth is already running with the same data directory '%v')", err, ctx.GlobalString(utils.DataDirFlag.Name))
+		}
```

Comment:
This still doesn't work. 

`ethdb.NewLDBDatabase` will return a `syscall.Errno` error for which you can check. In backend.go these errors are converted to an `*errors.errorString` instance. 

```
chainDb, err := newdb(filepath.Join(config.DataDir, "chaindata"))
if err != nil {
	return nil, fmt.Errorf("blockchain db err: %v", err)
}
```
The check in main.go for an errno wil then obviously fail.

Second problem is that the check if only added to the run subcommand. Users who will start the `console` (or any other subcommand) will still get the old error message when another instance already claimed the database datadir.